src/main/java/org/abzal1/ApplicationContextConfiguration.java

```diff
@@ -0,0 +1,55 @@
+package org.abzal1;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.*;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+
+import javax.sql.DataSource;
+import java.sql.*;
+
+@Configuration
+@ComponentScan(basePackages = "org.abzal1")
+@PropertySource("classpath:config.properties")
+public class ApplicationContextConfiguration {
+
+    @Value("${db.url}")
+    private String url;
+
+    @Value("${db.user}")
+    private String username;
+
+    @Value("${db.password}")
+    private String password;
+
+    @Value("${db.driverClassName}")
+    private String driverClassName;
+
+    @Bean
+    @Scope("singleton")
+    public DataSource dataSource() {
+        DriverManagerDataSource dataSource = new DriverManagerDataSource();
+        dataSource.setDriverClassName(driverClassName);
+        dataSource.setUrl(url);
+        dataSource.setUsername(username);
+        dataSource.setPassword(password);
+        return dataSource;
+    }
+//
+//    @Bean
+//    public JdbcTemplate jdbcTemplate(DataSource dataSource) {
+//        return new JdbcTemplate(dataSource);
+//    }
+
+    @Bean
+    public Connection connection(DataSource dataSource) {
```

Comment:
you should have a new connection for every transaction. In you case only one connection is created for hole project